### ISSUE TITLE
Component schema property ordering regression in ApiSpec.to_yaml() method

### ISSUE THREAD

**Issue by vorticity** [marshmallow-code/apispec#768]
In `apispec-5.1.1` schema property ordering in the yaml output matched the order of fields in the marshmallow schema, so long as you added a Meta() class to the schema, with an ordered=True attribute. As of `apispec-5.2.0` the marshmallow schema field ordering is no longer respected - instead the schema property ordering is always alphabetical.

My workaround is to re-implement `YamlDumper` and `_represent_dict()` functionality that was in `yaml_utils` up to `apispec-5.1.1` with an additional `add_representer()` call to ensure **OrderedDict** and **dict** instances are both ordered correctly and use this instead of the `to_yaml()` method:

```
def ordered_yaml_dump(data, stream=None, Dumper=SafeDumper, **kwds):
    class OrderedDumper(Dumper):
        pass

    def _represent_dict(dumper, data):
        return dumper.represent_mapping(yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG, data.items())

    OrderedDumper.add_representer(OrderedDict, _dict_representer)
    OrderedDumper.add_representer(dict, _dict_representer)
    return yaml.dump(data, stream, OrderedDumper, **kwds)
```
Related commit: https://github.com/marshmallow-code/apispec/commit/09dd98f26f75afefe17f22539dbedc7d64e5cb2b

**Comment by vorticity** [comment]
Some more testing reveals that the schema property ordering is respected and matches the output from the **OrderedDumper** if you set `sort_keys=False` in `yaml_dump_kwargs` ie:

`spec.to_yaml(yaml_dump_kwargs={"sort_keys": False})`

In either case, the output of `ApiSpec.to_yaml()` in the current version differs from the output of the same method in `apispec-5.1.1`

**Comment by lafrech** [comment]
Thanks for reporting and investigating.

What about this?

```py
def dict_to_yaml(dic, yaml_dump_kwargs=None):
    # By default, don't order alphabetically to respect schema order
    yaml_dump_kwargs.setdefault("sort_keys", False)
    if yaml_dump_kwargs is None:
        yaml_dump_kwargs = {}
    return yaml.dump(dic, **yaml_dump_kwargs)
```

Would you like to send a PR?

**Comment by vorticity** [comment]
Thanks for the quick response, here is the PR: https://github.com/marshmallow-code/apispec/pull/769

**Comment by lafrech** [comment]
Closed by #769.

And released.

Thanks.
